### PR TITLE
Fix POSTGIS

### PR DESCRIPTION
**Pull request by ebocher**
This PR fixes all POSTGIS tests
Linked to https://github.com/orbisgis/h2gis/pull/1087

### CODE REVIEW

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcSpatialTable.java [hunk 1]
```diff
@@ -200,19 +200,8 @@ public Envelope getExtend() {
                     LOGGER.error("Unable to get the ResultSet.");
                     return null;
                 }
-                List<String> names = new ArrayList<>(GeometryTableUtilities.getGeometryColumnNames(rs0.getMetaData()));
-                if (names.isEmpty()) {
-                    LOGGER.error("There is no geometric field.");
-                    return null;
-                }
+                return GeometryTableUtilities.getEnvelope(rs0);
```
[the remark this answers is not in the document]
Comment (SPalominos) [reply data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcSpatialTable.java:203]: Ok, but the message of H2GIS : `"The query doesn't contain any geometry field"` isn't clear as the user is not manipulation a query but a table.

File: data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ISpatialTable.java [hunk 2]
```diff
@@ -137,7 +137,7 @@
      * @return The full extend {@link Geometry} of the first geometry column of the table.
      */
     @Nullable
-    Envelope getExtend();
+    Geometry getExtend();
```
[the remark this answers is not in the document]
Comment (SPalominos) [reply data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ISpatialTable.java:140]: My bad!

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 3]
```diff
@@ -73,8 +72,8 @@ private static DriverFunction getDriverFromFile(@NotNull File file) {
             case "gpx":
                 return new GPXDriverFunction();
             default:
-                LOGGER.error("Unsupported file format.\n" +
-                        "Supported formats are : [shp, geojson, tsv, csv, dbf, kml, kmz, osm, gz, bz, gpx].");
+                LOGGER.error("Unsupported file format.\n"
+                        + "Supported formats are : [shp, geojson, tsv, csv, dbf, kml, kmz, osm, gz, bz, gpx].");
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:76]: Please keep the `+` a the end of the line.

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcSpatialTable.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ebocher) [reply data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcSpatialTable.java:203]: Yes but if we want to overwrite this message we need to get two times the geometry metadata.
So the solution will be to extract the H2gis code

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 4]
```diff
@@ -91,13 +90,13 @@ private static String unsupportedEncoding(@Nullable String encoding) {
      * Save a table to a file
      *
      * @param connection The connection to use for the save.
-     * @param tableName  Name of the table to save.
-     * @param filePath   Path of the destination file.
-     * @param encoding   Encoding of the file.
+     * @param tableName Name of the table to save.
+     * @param filePath Path of the destination file.
+     * @param encoding Encoding of the file.
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:95]: Please keep the `@param` description aligned.

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 5]
```diff
@@ -127,24 +126,24 @@ public static boolean saveAsFile(@NotNull Connection connection, @NotNull String
...
      * Load a file to a H2GIS database
      *
-     * @param filePath   the path of the file
-     * @param tableName  the name of the table created to store the file
-     * @param encoding   an encoding value to read the file
-     * @param deleteTable     true to delete the table if exists
+     * @param filePath the path of the file
+     * @param tableName the name of the table created to store the file
+     * @param encoding an encoding value to read the file
+     * @param deleteTable true to delete the table if exists
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:132]: Please keep the `@param` description aligned.

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 6]
```diff
@@ -127,24 +126,24 @@ public static boolean saveAsFile(@NotNull Connection connection, @NotNull String
...
      * @param dataSource the database
      */
     public static boolean loadFile(@NotNull String filePath, @NotNull String tableName, @Nullable String encoding,
-                                   boolean deleteTable, @NotNull JdbcDataSource dataSource) {
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java]: Please keep the parameters aligned

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 7]
```diff
@@ -91,13 +90,13 @@ private static String unsupportedEncoding(@Nullable String encoding) {
...
-     * @param tableName  Name of the table to save.
-     * @param filePath   Path of the destination file.
-     * @param encoding   Encoding of the file.
+     * @param tableName Name of the table to save.
+     * @param filePath Path of the destination file.
+     * @param encoding Encoding of the file.
      * @return True if the file has been saved, false otherwise.
      */
     public static boolean saveAsFile(@NotNull Connection connection, @NotNull String tableName,
-                                     @NotNull String filePath, @Nullable String encoding, @Nullable boolean deleteFile) {
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java]: Please keep the parameters aligned

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 8]
```diff
@@ -157,14 +156,14 @@ public static boolean loadFile(@NotNull String filePath, @NotNull String tableNa
...
      *
-     * @param properties      external database properties to set up the connection
-     * @param inputTableName  the name of the table in the external database
+     * @param properties external database properties to set up the connection
+     * @param inputTableName the name of the table in the external database
      * @param outputTableName the name of the table in the H2GIS database
-     * @param delete          true to delete the table if exists
-     * @param jdbcDataSource  the database
+     * @param delete true to delete the table if exists
+     * @param jdbcDataSource the database
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:163]: Please keep the `@param` description aligned

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcSpatialTable.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ebocher) [reply data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcSpatialTable.java:203]: I will do a fix

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java [hunk 9]
```diff
@@ -215,9 +214,9 @@ public static void loadTable(Map<String, String> properties, String inputTableNa
     /**
      * Create a dynamic link from a file
      *
-     * @param filePath       the path of the file
-     * @param tableName      the name of the table created to store the file
-     * @param delete         true to delete the table if exists
+     * @param filePath the path of the file
+     * @param tableName the name of the table created to store the file
+     * @param delete true to delete the table if exists
```
Comment (SPalominos) [inline data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:219]: Please keep the `@param` description aligned

File: data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java (hunk 3 above)
Comment (ebocher) [reply data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:76]: Automatic format put the + on the next line

Comment (SPalominos) [reply data-manager/jdbc/src/main/java/org/orbisgis/orbisdata/datamanager/jdbc/io/IOMethods.java:76]: all the code has the + at the end of the line

**Pull request merged**